spec.emu

```diff
@@ -322,17 +329,18 @@ contributors: Ujjwal Sharma, Younies Mahmoud
                     1. Set _nextValue_ to _nextValue_ + _duration_.[[Milliseconds]] / 10<sup>3</sup> + _duration_.[[Microseconds]] / 10<sup>6</sup> + _duration_.[[Nanoseconds]] / 10<sup>9</sup>.
                 1. If _nextValue_ is not 0 or _nextDisplay_ is not *"auto"*, then
                   1. Let _separator_ be _dataLocaleData_.[[digitalFormat]].[[separator]].
-                  1. Append the new Record { [[Type]]: *"literal"*, [[Value]]: _separator_} to the end of _result_.
+                  1. Append the new Record { [[Type]]: *"literal"*, [[Value]]: _separator_} to the end of _last_.
+                  1. Set _separated_ to *true*.
             1. Else,
               1. Perform ! CreateDataPropertyOrThrow(_nfOpts_, *"style"*, *"unit"*).
               1. Perform ! CreateDataPropertyOrThrow(_nfOpts_, *"unit"*, _numberFormatUnit_).
               1. Perform ! CreateDataPropertyOrThrow(_nfOpts_, *"unitDisplay"*, _style_).
               1. Let _nf_ be ! Construct(%NumberFormat%, &laquo; _durationFormat_.[[Locale]], _nfOpts_ &raquo;).
+              1. Let _list_ be a new empty List.
               1. Let _parts_ be ! PartitionNumberPattern(_nf_, 𝔽(_value_)).
-              1. Let _concat_ be an empty String.
-              1. For each Record { [[Type]], [[Value]], [[Unit]] } _part_ in _parts_, do
-                1. Set _concat_ to the string-concatenation of _concat_ and _part_.[[Value]].
-              1. Append the new Record { [[Type]]: _unit_, [[Value]]: _concat_ } to the end of _result_.
+              1. For each Record { [[Type]], [[Value]] } _part_ in _parts_, do
```

Comment:
I think `_parts_` is not defined in this scope

Reply:
`_parts_` is defined in the unchanged line above